### DISCUSSION TITLE
Dynamic VO Selection and Advanced Login Options

### DISCUSSION THREAD

**Discussion by aldbr** [DIRACGrid/diracx-web#14] [User personas and stories]
### Link to User Story

https://github.com/DIRACGrid/diracx-web/discussions/13

### Problem with Current Approach

The current login page lacks dynamic options. Users cannot select a VO during login, and there are no advanced options available for setting the group, specific properties, and API hostname. This makes the login process static and less user-friendly, especially for advanced users.

### Proposed Design Solution

1. **Dynamic VO Selection:** 
   - Add an option on the landing page for users to select a VO during login.
   - The available VOs can be dynamically computed from the vo:* scopes in `/.well-known/openid-configuration`

iguration.

2. **Advanced Login Options:**
   - Add hidden advanced options that users can expand if needed.
   - These options include setting the group, specific properties, and possibly even the API hostname for more flexibility.
   - This will be particularly useful for users who want to develop on localhost against a real installation.

3. **Mockup:**

Solution1:
- pros: nice if you have just a few VOs
- cons: not very scalable as you would need to scroll down at some point to find your VO
<img width="973" alt="Screenshot 2023-10-10 at 10 12 42(1)" src="https://github.com/DIRACGrid/diracx-web/assets/19813740/5384e27f-3e3a-4208-b822-7675e4d1d2b9">

Solution2:
- pros: scalable
- cons: more compact design
![image](https://github.com/DIRACGrid/diracx-web/assets/19813740/ed9c8499-344a-45b3-b469-8d31ddf762a9)

**Comment by chaen** [comment]
Conclusion: solution 1
